test/simulator_tests/birth_death_simulator_test.py

```diff
@@ -0,0 +1,530 @@
+        birth_waiting_dist = lambda _: 1
+        birth_scale_param = 1
+
+        tree = bd_sim.simulate_tree(
+            birth_waiting_dist, birth_scale_param, num_extant=32
+        )
+
+        results = test_tree(tree.get_tree_topology())
+        for i in results[0]:
+            self.assertEqual(i, 6)
+        self.assertEqual(results[1], 32)
+        self.assertTrue(results[2])
+
+        tree = bd_sim.simulate_tree(
+            birth_waiting_dist, birth_scale_param, experiment_time=6
+        )
+
+        results = test_tree(tree.get_tree_topology())
+        for i in results[0]:
+            self.assertEqual(i, 6)
+        self.assertEqual(results[1], 32)
+        self.assertTrue(results[2])
+
+    def test_nonconstant_yule(self):
+        bd_sim = BirthDeathFitnessSimulator()
+        birth_waiting_dist = lambda scale: np.random.exponential(scale)
+        birth_scale_param = 1
+
+        tree = bd_sim.simulate_tree(
+            birth_waiting_dist, birth_scale_param, num_extant=16
+        )
+
+        tree_top = tree.get_tree_topology()
+        results = test_tree(tree_top)
+        self.assertTrue(all(np.isclose(x, results[0][0]) for x in results[0]))
+        self.assertEqual(results[1], 16)
+        self.assertTrue(results[2])
+        self.assertEqual(max(tree.nodes), 31)
+
+        tree = bd_sim.simulate_tree(
+            birth_waiting_dist, birth_scale_param, experiment_time=2
+        )
+
+        tree_top = tree.get_tree_topology()
+        results = test_tree(tree_top)
+        for i in results[0]:
+            self.assertEqual(i, 2)
+        self.assertTrue(results[2])
+
+    def test_nonconstant_birth_death(self):
+        bd_sim = BirthDeathFitnessSimulator()
+        np.random.seed(1234)
+        birth_waiting_dist = lambda scale: np.random.exponential(scale)
+        death_waiting_dist = lambda: np.random.exponential(1.5)
+        birth_scale_param = 0.5
+
+        tree = bd_sim.simulate_tree(
+            birth_waiting_dist,
+            birth_scale_param,
+            death_waiting_dist,
+            num_extant=8,
+        )
+
+        tree_top = tree.get_tree_topology()
+        results = test_tree(tree_top)
+        self.assertTrue(all(np.isclose(x, results[0][0]) for x in results[0]))
+        self.assertEqual(results[1], 8)
+        self.assertTrue(results[2])
+        self.assertNotIn(9, tree_top.nodes)
+        self.assertNotIn(2, tree_top.nodes)
+
+        np.random.seed(1234)
+        tree = bd_sim.simulate_tree(
+            birth_waiting_dist,
+            birth_scale_param,
+            death_waiting_dist,
+            experiment_time=2,
+        )
+
+        tree_top = tree.get_tree_topology()
+        results = test_tree(tree_top)
+        for i in results[0]:
+            self.assertTrue(np.isclose(i, 2))
+        self.assertTrue(results[2])
+        self.assertNotIn(10, tree_top.nodes)
+        self.assertNotIn(2, tree_top.nodes)
+
+    def test_nonconstant_yule_with_predictable_fitness(self):
```

Comment:
Can we add a docstring (even if small / one liner) to the tests just to understand at a high level what is being tested in each function?